From c0ba7ab4bcd8e026f1afaf5d4950b50bbdbb3bc0 Mon Sep 17 00:00:00 2001
From: avandeve <arjan@linux.intel.com>
Date: Sat, 29 Mar 2014 20:45:08 -0700
Subject: [PATCH] When GPIO.s wakeup attribute is set, it.s supposed to wake
 the system up from

system sleep state, we should keep the GPIO.s IRQ enabled during suspend. So

IRQF_NO_SUSPEND flag is needed when request IRQ.

Change-Id: I7a9464ee42f934a84acf6bbdbe83984dac71dd8c
Signed-off-by: Aubrey Li <aubrey.li@intel.com>
---
 drivers/input/keyboard/gpio_keys.c |    2 ++
 1 file changed, 2 insertions(+)

Index: kernel-starpeak-3.14/drivers/input/keyboard/gpio_keys.c
===================================================================
--- kernel-starpeak-3.14.orig/drivers/input/keyboard/gpio_keys.c	2014-04-03 11:31:09.701400104 +0200
+++ kernel-starpeak-3.14/drivers/input/keyboard/gpio_keys.c	2014-04-03 11:31:09.689400104 +0200
@@ -502,6 +502,8 @@
 	 */
 	if (!button->can_disable)
 		irqflags |= IRQF_SHARED;
+	if (button->wakeup)
+		irqflags |= IRQF_NO_SUSPEND;
 
 	error = request_any_context_irq(bdata->irq, isr, irqflags, desc, bdata);
 	if (error < 0) {
